plugins/woocommerce/src/Utilities/ArrayUtil.php

```diff
@@ -286,40 +284,66 @@ private static function deep_compute_or_compare_array_diff( array $array1, array
 	/**
 	 * Push a value to an array, but only if the value isn't in the array already.
 	 *
-	 * @param array $array The array.
+	 * @param array $items The array.
 	 * @param mixed $value The value to maybe push.
 	 * @return bool True if the value has been added to the array, false if the value was already in the array.
 	 */
-	public static function push_once( array &$array, $value ) : bool {
-		if ( in_array( $value, $array, true ) ) {
+	public static function push_once( array &$items, $value ): bool {
+		if ( in_array( $value, $items, true ) ) {
 			return false;
 		}
 
-		$array[] = $value;
+		$items[] = $value;
 		return true;
 	}
 
 	/**
 	 * Ensure that an associative array has a given key, and if not, set the key to an empty array.
 	 *
-	 * @param array  $array The array to check.
+	 * @param array  $items The array to check.
 	 * @param string $key The key to check.
 	 * @param bool   $throw_if_existing_is_not_array If true, an exception will be thrown if the key already exists in the array but the value is not an array.
 	 * @return bool True if the key has been added to the array, false if not (the key already existed).
 	 * @throws \Exception The key already exists in the array but the value is not an array.
 	 */
-	public static function ensure_key_is_array( array &$array, string $key, bool $throw_if_existing_is_not_array = false ): bool {
-		if ( ! isset( $array[ $key ] ) ) {
-			$array[ $key ] = array();
+	public static function ensure_key_is_array( array &$items, string $key, bool $throw_if_existing_is_not_array = false ): bool {
+		if ( ! isset( $items[ $key ] ) ) {
+			$items[ $key ] = array();
 			return true;
 		}
 
-		if ( $throw_if_existing_is_not_array && ! is_array( $array[ $key ] ) ) {
-			$type = is_object( $array[ $key ] ) ? get_class( $array[ $key ] ) : gettype( $array[ $key ] );
+		if ( $throw_if_existing_is_not_array && ! is_array( $items[ $key ] ) ) {
+			$type = is_object( $items[ $key ] ) ? get_class( $items[ $key ] ) : gettype( $items[ $key ] );
+            // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
 			throw new \Exception( "Array key exists but it's not an array, it's a {$type}" );
 		}
 
 		return false;
 	}
-}
 
+	/**
+	 * Given an array of associative arrays, all having a shared key name ("column"), generates a new array in which
+	 * keys are the distinct column values found, and values are arrays with all the matches found
+	 * (or only the last matching array found, if $single_values is true).
+	 * See ArrayUtilTest for examples.
+	 *
+	 * @param array  $items The array to process.
+	 * @param string $column The name of the key to group by.
+	 * @param bool   $single_values True to only return the last suitable array found for each column value.
+	 * @return array The grouped array.
+	 */
+	public static function group_by_column( array $items, string $column, bool $single_values = false ): array {
```

Comment:
Does [wp_list_pluck](https://developer.wordpress.org/reference/functions/wp_list_pluck/) do basically the same thing?

Reply:
Nevermind, I see from the examples in the unit tests that they're different.